packages/web-pkg/src/composables/webWorkers/deleteWorker/worker.ts

```diff
@@ -0,0 +1,76 @@
+import PQueue from 'p-queue'
+import { Resource, encodePath } from '@ownclouders/web-client'
+import { WorkerTopics } from '../../piniaStores/webWorkers'
+
+type MessageData = {
+  baseUrl: string
+  accessToken: string
+  headers: Record<string, string>
+  resources: Resource[]
+}
+
+type Message = {
+  topic: WorkerTopics
+  data: MessageData
+}
+
+let accessToken: string
+
+self.onmessage = async (e: MessageEvent) => {
+  const { topic, data } = JSON.parse(e.data) as Message
+
+  if (accessToken === undefined || accessToken === null) {
+    // init access token, empty string if in public context
+    accessToken = data.accessToken || ''
+  }
+
+  if (topic === 'tokenUpdate') {
+    // token must only be updated for bearer tokens, not on public links
+    if (accessToken?.startsWith('Bearer')) {
+      accessToken = data.accessToken
+    }
+
+    return
+  }
+
+  const { baseUrl, headers, resources } = data
+
+  const successful: Resource[] = []
+  const failed: { resource: Resource; status: number }[] = []
+  const queue = new PQueue({ concurrency: 2 })
+
+  const doDelete = async (path: string) => {
+    // fake timer so requests take more time
+    await new Promise((res) => setTimeout(res, 2000))
+    return fetch(`${baseUrl}/${encodePath(path)}`, {
```

Comment:
Using our `web-client` here is a problem, right? Could you remind me why? I don't remember... but using raw `fetch` seems like a huge step back in DX. :-(

Reply:
Using the `ClientService` doesn't work because it needs to be initiated with so much stuff. Using the `web-client` could work though... I didn't think of that. It probably comes down to the question if we can use axios here. I'll give it a try 👍

Reply:
That would be nice. Thank you for checking it! 🤩

Reply:
It actually works 🙌  Feels a lot better this way!

Reply:
😍